Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -525,13 +525,33 @@ CGImageRef CGImageCreateCopyWithColorSpace(CGImageRef ref, CGColorSpaceRef color
 }
 
 /**
- @Status Stub
+ @Status Interoperable
+ @Notes This function does not defer the composition of its mask.
+        This function supports non-grayscale alpha masks, unlike the reference platform.
  */
 CGImageRef CGImageCreateWithMask(CGImageRef image, CGImageRef mask) {
-    // TODO  #1425 - masks are applied during rendering via D2D.
     RETURN_NULL_IF(!image);
-    UNIMPLEMENTED();
-    return StubReturn();
+    RETURN_NULL_IF(!mask);
+    RETURN_NULL_IF(CGImageIsMask(image));
+
+    size_t width = CGImageGetWidth(image);
+    size_t height = CGImageGetHeight(image);
+
+    woc::unique_cf<CGContextRef> context{ CGBitmapContextCreate(nullptr,
```

Comment:
Yep, I'm thinking if someone decides to change the bitmap info (without realizing we don't support all of it due to D2D restrictions), we would want to catch that failure fast.

---
In reply to: [92708281](https://github.com/Microsoft/WinObjC/pull/1555#discussion_r92708281) [](ancestors = 92708281)